tunnel.go

```diff
@@ -16,29 +17,37 @@ import (
 
 // Tunnel describes a ssh-like tunnel to a kubernetes pod.
 type Tunnel struct {
-	Local     int
-	Remote    int
-	Namespace string
-	PodName   string
-	Out       io.Writer
-	stopChan  chan struct{}
-	readyChan chan struct{}
-	config    *rest.Config
-	client    rest.Interface
+	Local        int
+	Remote       int
+	Namespace    string
+	ResourceName string
+	ResourceType string
+	Out          io.Writer
+	stopChan     chan struct{}
+	readyChan    chan struct{}
+	config       *rest.Config
+	client       rest.Interface
 }
 
 // NewTunnel creates a new tunnel.
-func NewTunnel(client rest.Interface, config *rest.Config, namespace, podName string, remote int) *Tunnel {
-	return &Tunnel{
-		config:    config,
-		client:    client,
-		Namespace: namespace,
-		PodName:   podName,
-		Remote:    remote,
-		stopChan:  make(chan struct{}, 1),
-		readyChan: make(chan struct{}, 1),
-		Out:       ioutil.Discard,
+func NewTunnel(client rest.Interface, config *rest.Config, namespace, resource string, remote int) (*Tunnel, error) {
```

Comment:
When I see this I think it would be cool to actually have a proper config struct.